fcrepo-stats-impl/src/main/java/org/fcrepo/stats/impl/DbStatsImpl.java

```diff
@@ -0,0 +1,113 @@
+
+    /**
+     * Setup template
+     */
+    @PostConstruct
+    public void setup() {
+        this.jdbcTemplate = new NamedParameterJdbcTemplate(this.dataSource);
+    }
+
+    @Override
+    public StatsResults getStatistics(final StatsParameters statsParams) {
+        final var results = new StatsResults();
+        //count all resources
+        final var resourceCount = jdbcTemplate.queryForObject(SELECT_COUNT_FROM_SIMPLE_SEARCH,
+                new MapSqlParameterSource(),
+                Long.class);
+
+        final var all = new StatsResult();
+        all.setResourceCount(resourceCount);
+        results.setAll(all);
+
+        final var mimeTypes = statsParams.getMimeTypes();
+        final var parameterSource = new MapSqlParameterSource();
+        final var mimeTypesQuery = formatMimetypeQuery(mimeTypes, parameterSource);
+        //count binaries by mimetype
+        final var mimetypeResults = jdbcTemplate.queryForRowSet(mimeTypesQuery, parameterSource);
+        marshallMimeTypeResults(results, mimetypeResults);
+        return results;
+    }
+
+    private void marshallMimeTypeResults(final StatsResults results, final SqlRowSet mimetypeResults) {
+        if (mimetypeResults.first()) {
+            final var mimeTypesResultList = new ArrayList<MimetypeStatsResult>();
+            do {
+                final var mimetypeResult = new MimetypeStatsResult();
+                mimetypeResult.setMimetype(mimetypeResults.getString(1));
+                mimetypeResult.setResourceCount(mimetypeResults.getLong(2));
+                mimetypeResult.setByteCount(mimetypeResults.getLong(3));
+                mimeTypesResultList.add(mimetypeResult);
+            } while (mimetypeResults.next());
+            results.setMimetypes(mimeTypesResultList);
+
+            //sum the mimetype counts for all binaries - no need to perform a separate query
+            final var binary = new BinaryStatsResult();
+            final AtomicLong byteCount = new AtomicLong(0);
+            final AtomicLong binaryResourceCount = new AtomicLong(0);
+            mimeTypesResultList.forEach(x -> {
+                byteCount.addAndGet(x.getByteCount());
+                binaryResourceCount.addAndGet(x.getResourceCount());
+            });
+
+            binary.setResourceCount(binaryResourceCount.get());
+            binary.setByteCount(byteCount.get());
+            results.setBinaries(binary);
+        }
+    }
+
+    private String formatMimetypeQuery(final List<String> mimeTypes, final MapSqlParameterSource parameterSource) {
+        final var mimeTypesQuery = new StringBuilder("select a.mime_type, count(a.*), sum(a.content_size) ");
+        mimeTypesQuery.append("from simple_search a, search_resource_rdf_type b,  search_rdf_type c ");
+        mimeTypesQuery.append("where a.id = b.resource_id and b.rdf_type_id = c.id and c.rdf_type_uri = '");
+        mimeTypesQuery.append(RdfLexicon.NON_RDF_SOURCE.getURI());
```

Comment:
Is there any possibility this might need to be escaped?

Reply:
I don't think so.